### PR TITLE
feat(integrations): add support for jobdiva

### PR DESCRIPTION
**Pull request by hassan254-prog**
Describe the problem and your solution

- Add support for Jobdiva

---

**JobDiva Integration: Provider, Auth Flow, Documentation, and Token Parsing Enhancements**

This PR introduces first-class support for the JobDiva provider in the Nango platform. Key changes include a new provider definition for JobDiva with a non-standard, non-expiring token-based authentication flow, updates to TypeScript typings for token expiration handling, adjustments in token parsing utilities, and new user-facing documentation (including setup guides and branding). The PR also ensures robust handling of JobDiva's unique token response shape within token extraction logic and expands test coverage for edge-case timestamp parsing. All affected areas, from codebase to docs and assets, are updated to accommodate JobDiva-specific requirements.

<details>
<summary><strong>Key Changes</strong></summary>

• Added `JobDiva` provider configuration in providers.yaml with custom credential fields and proxy settings
• Introduced `JobDiva` documentation and setup guides with navigation integration
• Added `JobDiva` ``SVG`` logo asset for ``UI`` branding
• Extended `ProviderTwoStep` typing: `token_expiration` and `token_expiration_strategy` are now optional
• Enhanced `parseRawCredentials` to support plain string/infinite-expiry token responses (`JobDiva`'s approach)
• Improved `parseTokenExpirationDate` utility for robust ms/seconds timestamp parsing
• Expanded utility test suite for millisecond-based expiration handling
• Documented pre-built tooling and integration status
• Updated docs navigation (docs.json) for `JobDiva` inclusion

</details>

<details>
<summary><strong>Affected Areas</strong></summary>

• Provider configuration (providers.yaml)
• Credential/token parsing utilities and handling (shared/lib/services/connection.service.ts, shared/lib/utils/utils.ts)
• `TypeScript` typings (`ProviderTwoStep` type in types/lib/providers/provider.ts)
• Documentation and setup guides (docs-v2/integrations/all/jobdiva.*)
• Frontend image assets (webapp/public/images/template-logos/)
• Test coverage for token/date parsing (shared/lib/utils/utils.unit.test.ts)
• Docs v2 navigation (docs-v2/docs.json)

</details>

---
*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/providers/providers.yaml [hunk 1]
```diff
@@ -6057,6 +6057,41 @@ jobadder:
...
+    display_name: JobDiva
+    categories:
+        - hr
+    auth_mode: TWO_STEP
+    token_url: https://api.jobdiva.com/apiv2/authenticate?clientid=${credentials.clientId}&username=${credentials.username}&password=${credentials.password}
+    token_request_method: GET
+    proxy:
+        base_url: https://api.jobdiva.com
+    token_response:
+        token:
```
Comment (hassan254-prog) [inline packages/providers/providers.yaml]: They return a plain string in the token request response.

File: packages/shared/lib/utils/utils.ts [hunk 2]
```diff
@@ -79,6 +81,11 @@ export function parseTokenExpirationDate(expirationDate: any): Date | undefined
         return expirationDate;
     }
 
+    // Handle "99years"
+    if (expirationDate === DEFAULT_INFINITE_EXPIRES_AT_MS) {
+        return new Date(Date.now() + DEFAULT_INFINITE_EXPIRES_AT_MS);
+    }
```
Comment (bodinsamuel) [inline packages/shared/lib/utils/utils.ts]: I would make that DEFAULT_INFINITE_EXPIRES_AT_MS is always a date in your new scenario 
or
like the ms/s situation for the headers, handle that in the if === number and

Comment (hassan254-prog) [reply packages/shared/lib/utils/utils.ts]: Thanks, updated with the latest [commit](https://github.com/NangoHQ/nango/pull/4514/commits/05f823d9beeaea7cd672a63cad7167b13a3e31aa).

File: packages/shared/lib/utils/utils.ts [hunk 3]
```diff
@@ -79,8 +79,15 @@ export function parseTokenExpirationDate(expirationDate: any): Date | undefined
...
 
-    // UNIX timestamp
     if (typeof expirationDate === 'number') {
+        const dateFromValue = new Date(expirationDate);
+        // Check if expirationDate is in milliseconds
+        const isMs = dateFromValue.getFullYear() > 1971;
+        const currentTime = Date.now();
+        if (isMs) {
+            return new Date(currentTime + expirationDate);
+        }
```
Comment (bodinsamuel) [inline packages/shared/lib/utils/utils.ts:88]: Are you sure about this logic?

When we receive expirationDate, it's a fully fledged timestamp, no? So why would you add currentTime on top of it?

Comment (hassan254-prog) [reply packages/shared/lib/utils/utils.ts:88]: yes 🤦‍♂️, thanks for catching that, guess it was working since its 99 years,  I think its better to change the ms to a full timestamp before. I have updated that with the latest [commit](https://github.com/NangoHQ/nango/pull/4514/commits/f86f72226506db733c97bd5d607f6b09bc50d052).

**Review by bodinsamuel — APPROVED** [review]
